### PR TITLE
Add tomcat regression test

### PR DESCRIPTION
**Pull request by ggkioulis**
Summary: Tomcat regression test
 * install and configure tomcat
 * access tomcat manager web page
 * test all Servlet, JSP and WebSocket examples

- Related ticket: https://progress.opensuse.org/issues/50060
- Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/1215
- Verification run:
sle 12.sp2: http://10.161.229.232/tests/546#step/tomcat/1
sle 12.sp3: http://10.161.229.232/tests/542#step/tomcat/1
sle 12.sp4: http://10.161.229.232/tests/517#step/tomcat/1
sle 12.sp5: http://10.161.229.232/tests/708#step/tomcat/1
sle 15: http://10.161.229.232/tests/683#step/tomcat/1
sle 15.sp1: http://10.161.229.232/tests/689#step/tomcat/1
sle 15.sp2: http://10.161.229.232/tests/714#step/tomcat/1

### CODE REVIEW

File: lib/Tomcat/JspTest.pm [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+    # test xhtml svg example
+    $self->firefox_open_url('localhost:8080/examples/jsp/jsp2/jspx/textRotate.jspx?name=testing');
+    send_key_until_needlematch('tomcat-xhtml-svg', 'ret');
+}
+
+
+# test basic arithmetics example
+sub basic_arithmetics_example() {
+    assert_screen('tomcat-jsp-basic-arithmetics', 60);
```
Comment (asmorodskyi) [inline lib/Tomcat/JspTest.pm]: please replace timeout with variable

File: lib/Tomcat/Utils.pm [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    select_console('root-console');
+
+    record_info('Initial Setup');
+
+    assert_script_run "SUSEConnect --status-text |grep -A3 -E 'SUSE Linux Enterprise Server|Web and Scripting Module' | grep -qE '^\\s+Registered'";
+    # enable Web and Scripting Module for SLE 15 and above
+    if (is_sle('>=15')) {
+        add_suseconnect_product("sle-module-web-scripting");
+    }
+    assert_script_run "SUSEConnect --status-text |grep -A3 -E 'SUSE Linux Enterprise Server|Web and Scripting Module' | grep -qE '^\\s+Registered'";
```
Comment (asmorodskyi) [inline lib/Tomcat/Utils.pm]: didn't get why you running same command twice ?

File: lib/main_common.pm [hunk 3]
```diff
@@ -1262,6 +1262,7 @@ sub load_x11tests {
         loadtest "x11/midori" unless (is_staging || is_livesystem);
         loadtest "x11/ristretto";
     }
+    loadtest "x11/tomcat" unless is_sle('<12-sp2');
```
Comment (asmorodskyi) [inline lib/main_common.pm]: this way you will also trigger it for SLE 12 SP5 and SLE 15 SP2 . Maybe it is good but need to check this or limit execution to already released versions

**Comment by asmorodskyi** [comment]
1. from pure perl language perspective I think it is pretty awesome solution ! I would record this trick for myself 
2. **but** from openQA testing perspective I would say this testing should not be done in openQA at all.  You introducing 130 needles . this mean you would need to **support** them . this is huge amount of work in case of any UI/ font  changes . On other side you can easily achieve same coverage by writing simple bash script which would `curl` over every mentioned URL. You may still execute it inside openQA 
3. About function naming - I have read somewhere nice rule - "if all function / variable names in a file has same starting / ending just cut it you won't loose anything "

File: lib/Tomcat/Utils.pm (hunk 2 above)
Comment (ggkioulis) [reply lib/Tomcat/Utils.pm]: ah thanks, for some reason I missed this when I was reviewing my code!

File: lib/main_common.pm (hunk 3 above)
Comment (ggkioulis) [reply lib/main_common.pm]: I think it is ok if it is triggered for those SLE versions as well, I would keep close attention to the runs on the first couple of weeks in order to fix any potential issues and make the test more robust.

File: lib/Tomcat/JspTest.pm (hunk 1 above)
Comment (ggkioulis) [reply lib/Tomcat/JspTest.pm]: done!

File: lib/Tomcat/Utils.pm [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+EOF
+
+    # log in to root console
+    select_console('root-console');
+
+    record_info('Initial Setup');
+
+    # enable Web and Scripting Module for SLE 15 and above
+    if (is_sle('>=15')) {
+        add_suseconnect_product("sle-module-web-scripting");
```
Comment (baierjan) [inline lib/Tomcat/Utils.pm]: I am still not sure about this line. Isn't the preferred method using `SCC_ADDONS=serverapp,script`?

Comment (ggkioulis) [reply lib/Tomcat/Utils.pm]: I 'm on it!

Comment (dzedro) [reply lib/Tomcat/Utils.pm]: I'm sure QAM tests have `SCC_ADDONS=serverapp,script` but tests on developed products not.

File: lib/main_common.pm (hunk 3 above)
Comment (dzedro) [reply lib/main_common.pm]: Why schedule it here ? allpatterns is alredy long enough (1:30), maybe better put it into webserver or extratests ?

File: lib/Tomcat/Utils.pm (hunk 4 above)
Comment (baierjan) [reply lib/Tomcat/Utils.pm]: So maybe using `unless is_released` could be the right way to go?

File: lib/main_common.pm (hunk 3 above)
Comment (dzedro) [reply lib/main_common.pm]: This will affect also staging

Comment (baierjan) [reply lib/main_common.pm]: Personally, I would go for `EXTRATEST=tomcat`

Comment (ggkioulis) [reply lib/main_common.pm]: > Personally, I would go for `EXTRATEST=tomcat`

so, you mean create a new extra_tests test suite for tomcat?

Comment (baierjan) [reply lib/main_common.pm]: Yes, in the light of current trends (splitting test suites), I would create `load_extra_tests_tomcat` and put it there.

**Pull request merged**